xchange-bitstamp/src/main/java/org/knowm/xchange/bitstamp/dto/account/BitstampEarnTransaction.java

```diff
@@ -0,0 +1,39 @@
+package org.knowm.xchange.bitstamp.dto.account;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.math.BigDecimal;
+import java.util.Date;
+import lombok.Getter;
+import lombok.ToString;
+import org.knowm.xchange.bitstamp.BitstampUtils;
+import org.knowm.xchange.currency.Currency;
+
+@Getter
+@ToString
+public class BitstampEarnTransaction {
+
+  private final Date datetime;
+  private final String type; // SUBSCRIBE, UNSUBSCRIBE, etc.
+  private final BigDecimal amount;
+  private final Currency currency;
+  private final BigDecimal value;
+  private final Currency quoteCurrency;
+  private final String status;
+
+  public BitstampEarnTransaction(
+      @JsonProperty("datetime") String datetime,
+      @JsonProperty("type") String type,
+      @JsonProperty("amount") BigDecimal amount,
+      @JsonProperty("currency") String currency,
+      @JsonProperty("value") BigDecimal value,
+      @JsonProperty("quote_currency") String quoteCurrency,
+      @JsonProperty("status") String status) {
+    this.datetime = datetime != null ? BitstampUtils.parseDate(datetime) : null;
+    this.type = type;
+    this.amount = amount;
+    this.currency = Currency.getInstance(currency);
```

Comment:
you can omit constructor and annotate fields with converter, see examples:

- `org.knowm.xchange.bitget.config.converter.StringToCurrencyConverter`
- `BitgetBalanceDto#currency'